src/collection-view.js

```diff
@@ -321,9 +337,19 @@ Marionette.CollectionView = Marionette.View.extend({
     // remove and/or close it later
     this.children.add(view);
 
+    if (canTriggerAttach && this.triggerBeforeAttach) {
+      nestedViews = [view].concat(view._getNestedViews());
+      // Trigger before:attach after render to honor Region#show() lifecycle order.
+      view.once('render', _.partial(this._triggerMethodOnAll, nestedViews, 'before:attach'), this);
```

Comment:
Hey @jasonLaster, final call on this?  I'm worried about putting a bunch of logic in renderChildView, which right now is so straightforward.  Not to mention logic based on a private variable.  From that point of view I think using a "once render" is not as messy.